Clarify VM Insights support with Azure Monitor Agent in Entra ID authentication documentation (#28)
MicrosoftDocs/azure-monitor-docs: articles/azure-monitor/logs/azure-ad-authentication-logs.md

- Existing Log Analytics agents will stop functioning. Only Azure Monitor Agent will be supported. Azure Monitor Agent will be missing some capabilities that are available through the Log Analytics agent. Examples include custom log collection and IIS log collection.
- The Data Collector API (preview) won't support Microsoft Entra authentication and won't be available to ingest data.

> - VM Insights will only be supported with the Azure Monitor Agent (AMA).

Reviewer: ```suggestion
- VM Insights is supported only with Azure Monitor Agent (AMA).
```
Minor edits for Microsoft style
